projects/storefrontlib/src/shared/components/form/form-errors/form-errors.component.ts

```diff
@@ -53,4 +51,45 @@ export class FormErrorsComponent {
   @HostBinding('class.control-touched') get touched() {
     return this.control?.touched;
   }
+
+  ngOnChanges(changes: SimpleChanges) {
+    if (changes.control) {
+      this.control = changes.control.currentValue;
+    } else if (changes.controlName) {
+      this.control = this.getFormGroup()?.get(
+        changes.controlName.currentValue
+      ) as FormControl;
+    }
+    this.setErrors();
+  }
+
+  /**
+   * get the form from the FormGroupDirective
+   * @private
+   */
+  private getFormGroup(): FormGroup {
+    return this._formGroup ? this._formGroup.form : null;
+  }
+
+  /**
+   * Get all errors of the control and show the first one only
+   * according to the order in the formControl declaration
+   * @private
+   */
+  private setErrors(): void {
+    this.errors$ = this.control?.statusChanges.pipe(
+      startWith({}),
+      map(() => this.control.errors || {}),
+      map((errors) => {
+        const filteredErrors = Object.entries(errors)
+          .filter((error) => error[1])
+          .map((error) => error[0]);
+        return this.showOnlyOne
+          ? filteredErrors[0]
```

Comment:
`filteredErrors?.[0]` ?

Reply:
The issue is that we need to wrap the `[0]` element in an array to be iterateable in the template.
Without the 2nd ternary operator ,`[undefined]` not an `[]` is returned.

Reply:
understood. you could do `filteredErrors?.[0] ?? []` i guess.